simpleclient_hotspot/src/main/java/io/prometheus/client/hotspot/ThreadExports.java

```diff
@@ -73,8 +77,42 @@ void addThreadMetrics(List<MetricFamilySamples> sampleFamilies) {
         "jvm_threads_deadlocked_monitor",
         "Cycles of JVM-threads that are in deadlock waiting to acquire object monitors",
         nullSafeArrayLength(threadBean.findMonitorDeadlockedThreads())));
+
+    GaugeMetricFamily threadStateFamily = new GaugeMetricFamily(
+      "jvm_threads_state",
+      "Current count of threads by state",
+      Collections.singletonList("state"));
+
+    Map<Thread.State, Integer> threadStateCounts = getThreadStateCountMap();
+    for (Thread.State state : Thread.State.values()) {
```

Comment:
It'd be a bit simpler to do this down in the function before fetching the data.

Reply:
Ok I'll move the code.